extensions/quick-access/src/search-pinned-directory.tsx

```diff
@@ -0,0 +1,240 @@
+import React, { useState } from "react";
+import { commonPreferences, isEmpty, isImage } from "./utils/common-utils";
+import { DirectoryInfo, DirectoryType, FileInfo } from "./utils/directory-info";
+import { parse } from "path";
+import { pinDirectory } from "./pin-directory";
+import { LocalStorageKey } from "./utils/constants";
+import { copyFileByPath } from "./utils/applescript-utils";
+import { alertDialog, copyLatestFile, localDirectoryWithFiles, refreshNumber } from "./hooks/hooks";
+import fse from "fs-extra";
+import { ActionRemoveAllDirectories } from "./utils/ui-components";
+
+export default function Command() {
+  const [tag, setTag] = useState<string>("All");
+  const [refresh, setRefresh] = useState<number>(0);
+  const { autoCopyLatestFile } = commonPreferences();
+
+  const { pinnedDirectory, directoryWithFiles, allFilesNumber, directoryTags, loading } =
+    localDirectoryWithFiles(refresh);
+  copyLatestFile(autoCopyLatestFile, directoryWithFiles);
+
+  return (
+    <List
+      isLoading={loading}
+      searchBarPlaceholder={"Search file"}
+      searchBarAccessory={
+        <List.Dropdown onChange={setTag} tooltip={"Directory type"}>
+          <List.Dropdown.Item key={"All"} title={"All Directory"} value={"All"} />
+          {directoryTags.map((value, index) => {
+            return <List.Dropdown.Item key={index + value} title={value} value={value} />;
+          })}
+        </List.Dropdown>
+      }
+    >
+      {directoryWithFiles.length === 0 || allFilesNumber === 0 ? (
+        <List.EmptyView
+          title={directoryWithFiles.length === 0 ? "No directory. Please pin first" : "No Files."}
+          description={"You can always pin directory from the Action Panel"}
+          actions={
+            <ActionPanel>
+              <Action
+                title={"Pin Directory"}
+                icon={Icon.Download}
+                onAction={async () => {
+                  await pinDirectory(false);
+                  setRefresh(refreshNumber());
+                }}
+              />
+              {directoryWithFiles.length != 0 && <ActionRemoveAllDirectories setRefresh={setRefresh} />}
+            </ActionPanel>
+          }
+        />
+      ) : (
+        <>
+          {directoryWithFiles.map(
+            (directory, directoryIndex) =>
+              (tag == directory.directory.name || tag == "All") && (
+                <List.Section
+                  key={directory.directory.id}
+                  title={directory.directory.name}
+                  subtitle={parse(directory.directory.path).dir}
+                >
+                  {directory.files.map((value, index) => (
+                    <List.Item
+                      key={value.id}
+                      icon={isImage(parse(value.path).ext) ? { source: value.path } : { fileIcon: value.path }}
+                      title={value.name}
+                      accessories={index === 0 ? [{ icon: "🌟" }] : []}
```

Comment:
This icon feels a lil out of place. If you decide to keep it, can you also add a `tooltip` to describe it?

Reply:
Maybe I can find a better way to describe it, and if I do, I'll change it; if not, I'll change 🌟 to "Latest File".